roadmap/implementers-guide/src/node/collators/collator-protocol.md

```diff
@@ -108,14 +108,20 @@ When acting on an advertisement, we issue a `Requests::CollationFetching`. If th
 
 As a validator, once the collation has been fetched some other subsystem will inspect and do deeper validation of the collation. The subsystem will report to this subsystem with a [`CollatorProtocolMessage`][CPM]`::ReportCollator` or `NoteGoodCollation` message. In that case, if we are connected directly to the collator, we apply a cost to the `PeerId` associated with the collator and potentially disconnect or blacklist it.
 
-### Interaction with [Candidate Selection][CS]
+### Interaction with [Candidate Backing][CB]
 
-As collators advertise the availability, we notify the Candidate Selection subsystem with a [`CandidateSelection`][CSM]`::Collation` message. Note that this message is lightweight: it only contains the relay parent, para id, and collator id.
+As collators advertise the availability, we notify the Candidate Selection subsystem with a [`CandidateBacking`][CBM]`::Second` message. Note that this message contains the relay parent of the advertised collation, the candidate receipt and the [PoV][PoV].
 
-At that point, the Candidate Selection algorithm is free to use an arbitrary algorithm to determine which if any of these messages to follow up on. It is expected to use the [`CollatorProtocolMessage`][CPM]`::FetchCollation` message to follow up.
+For the moment, a validator will simply second the first valid parablock candidate per relay head. See [Future Work](#future-work).
 
-The intent behind this design is to minimize the total number of (large) collations which must be transmitted.
+## Future Work
 
+Several approaches have been discussed, but all have some issues:
+
+- The current approach is very straightforward. However, that protocol is vulnerable to a single collator which, as an attack or simply through chance, gets its block candidate to the node more often than its fair share of the time.
+- It may be possible to do some BABE-like selection algorithm to choose an "Official" collator for the round, but that is tricky because the collator which produces the PoV does not necessarily actually produce the block.
+- We could use relay-chain BABE randomness to generate some delay `D` on the order of 1 second, +- 1 second. The collator would then second the first valid parablock which arrives after `D`, or in case none has arrived by `2*D`, the last valid parablock which has arrived. This makes it very hard for a collator to game the system to always get its block nominated, but it reduces the maximum throughput of the system by introducing delay into an already tight schedule.
+- A variation of that scheme would be to randomly choose a number `I`, and have a fixed acceptance window `D` for parablock candidates. At the end of the period `D`, count `C`: the number of parablock candidates received. Second the one with index `I % C`. Its drawback is the same: it must wait the full `D` period before seconding any of its received candidates, reducing throughput.
```

Comment:
I'd worry about us building DoS attacks against ourselves, ala https://github.com/paritytech/polkadot/issues/2203  Races sound dangerous. 

We've [one grant applicant](https://github.com/w3f/Open-Grants-Program/pull/357) with [storage ideas](https://drive.google.com/file/d/1v847u_XeVf0SBz7Y7LEMXi72QfqirstL/view) similar to my [mixnet incentives design](https://github.com/w3f/messaging/blob/master/incentives/incentives.tex).  Initially, they wanted their proof-of-useful-storage to actually produce blocks,  I suggested they use Aura, etc. but run a sampling procedure that rewards storage separately, which seems easier to control.